sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -442,7 +442,11 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   override def uncacheTable(tableName: String): Unit = {
```

Comment:
I don't think we should put dropping view at `uncacheTable`. `cacheTable` in the same `CatalogImpl` doesn't create view.

`CacheTableCommand` creates view for `cache table as select` query. So I think it is more correct to drop the view in `UncacheTableCommand`.

Reply:
I have also considered this issue. Finally, I decided to it in `uncacheTable` for two reasons.
1. `UncacheTableCommand` only uses `sparkSession.catalog.uncacheTable` to do the operation of deleting the cache, and there is no other operation.
2.  
There are many other places that use `sparkSession.catalog.uncacheTable` to implement the operation of `uncache table`.  I think these places also need to consider drop the view.

@viirya May I ask if these considerations are correct?

Reply:
If the view isn't created when caching, I don't think it is good to remove the view when uncaching. `CacheTableCommand` is the caching command creating the view. I think it makes more sense to remove the view at `UncacheTableCommand`. They are a pair of commands.

When you do deleting view at `uncacheTable`. For a view which isn't created by `cache table as select`, uncaching it will delete it too? This sounds not correct to me.

Reply:
@viirya Yes. There is indeed this problem. However, there is still a problem of accidental deletion when removing the view at `UncacheTableCommand`. Because when executing `uncache table`, I can't judge whether the temporary view is created by the user or created when the `cache table` is executed.

I want to ask, why create a view when executing `cache table as select` . What does this view do? Because there is no correspondence between `view `and `plan`.